### PR TITLE
Feat/msdk 339 get pending instructions refactor

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/api/entities/CurrentIdentity.ts [hunk 1]
```diff
@@ -82,11 +86,20 @@ export class CurrentIdentity extends Identity {
       context,
     } = this;
 
-    const [, ...numberedPortfolios] = await portfolios.getPortfolios();
+    const getPortfolios = await portfolios.getPortfolios();
```
Comment (monitz87) [inline src/api/entities/CurrentIdentity.ts:89]: ownedPortfolios

File: src/api/entities/CurrentIdentity.ts [hunk 2]
```diff
@@ -82,11 +86,20 @@ export class CurrentIdentity extends Identity {
...
+
+    const [ownedCustodiedPortfolios, custodiedPortfolios] = await Promise.all([
+      P.filter(getPortfolios, async portfolio => {
+        const { did: custodianIdentityDid } = await portfolio.getCustodian();
+        return custodianIdentityDid === did;
+      }),
```
Comment (monitz87) [inline src/api/entities/CurrentIdentity.ts:95]:
```suggestion
      P.filter(getPortfolios, portfolio => portfolio.isCustodiedBy(did)),
```

File: src/api/entities/CurrentIdentity.ts [hunk 3]
```diff
@@ -82,11 +86,20 @@ export class CurrentIdentity extends Identity {
...
+      this.portfolios.getCustodiedPortfolios(),
+    ]);
+
+    const allPortfolios = [...ownedCustodiedPortfolios, ...custodiedPortfolios];
 
     const portfolioIds: PortfolioId[] = [
-      { did },
-      ...numberedPortfolios.map(({ id: number }) => ({ did, number })),
+      ...(await P.map(allPortfolios, portfolio => portfolioLikeToPortfolioId(portfolio, context))),
     ];
```
Comment (monitz87) [inline src/api/entities/CurrentIdentity.ts:103]: You can just do `const portfolioIds = await P.map(allPortfolios, portfolio =>  portfolioLikeToPortfolioId(portfolio, context))`

File: src/api/entities/__tests__/CurrentIdentity.ts [hunk 4]
```diff
@@ -1,14 +1,28 @@
 import BigNumber from 'bignumber.js';
 import sinon from 'sinon';
 
-import { CurrentIdentity, Identity, Venue } from '~/api/entities';
+import { CurrentIdentity, Identity, NumberedPortfolio, Venue } from '~/api/entities';
+import { DefaultPortfolio } from '~/api/entities/DefaultPortfolio';
```
Comment (monitz87) [inline src/api/entities/__tests__/CurrentIdentity.ts:5]: This should be part of the above import

File: src/api/entities/__tests__/CurrentIdentity.ts [hunk 5]
```diff
@@ -1,14 +1,28 @@
...
+  require('~/testUtils/mocks/entities').mockNumberedPortfolioModule(
+    '~/api/entities/NumberedPortfolio'
+  )
+);
+jest.mock(
+  '~/api/entities/DefaultPortfolio',
+  require('~/testUtils/mocks/entities').mockDefaultPortfolioModule(
+    '~/api/entities/DefaultPortfolio'
+  )
+);
```
Comment (monitz87) [inline src/api/entities/__tests__/CurrentIdentity.ts:24]: Are you sure these are necessary?

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/entities/__tests__/CurrentIdentity.ts (hunk 5 above)
Comment (shuffledex) [reply src/api/entities/__tests__/CurrentIdentity.ts:24]: yes, they are (portfolioCustodian)

**Review by monitz87 — APPROVED** [review]

**Pull request merged**